provider/cloudflare/cloudflare.go

```diff
@@ -140,25 +140,71 @@ type RecordParamsTypes interface {
 }
 
 // getUpdateDNSRecordParam is a function that returns the appropriate Record Param based on the cloudFlareChange passed in
-func getUpdateDNSRecordParam(cfc cloudFlareChange) cloudflare.UpdateDNSRecordParams {
-	return cloudflare.UpdateDNSRecordParams{
+func getUpdateDNSRecordParam(cfc cloudFlareChange) (cloudflare.UpdateDNSRecordParams, error) {
+	recordParam := cloudflare.UpdateDNSRecordParams{
 		Name:    cfc.ResourceRecord.Name,
 		TTL:     cfc.ResourceRecord.TTL,
 		Proxied: cfc.ResourceRecord.Proxied,
 		Type:    cfc.ResourceRecord.Type,
-		Content: cfc.ResourceRecord.Content,
 	}
+
+	if cfc.ResourceRecord.Type == "SRV" {
+		recordParam.Data = cfc.ResourceRecord.Data
+	} else {
+		recordParam.Content = cfc.ResourceRecord.Content
+	}
+
+	return recordParam, nil
```

Comment:
I'm not sure to see when error can be != than nil in this method. 
Did I missed something ?

Reply:
@starcraft66 https://github.com/cloudflare/cloudflare-go/blob/b873d58b014cfe9a9538e7c0ce5bfa3ccfabfbde/dns.go#L177 doesn't export an error.

This function doesn't export an error, and currently will always return nil so this and lines https://github.com/kubernetes-sigs/external-dns/blob/8dd6fe6d0fe5bc6156a458763c481a611e3378f9/provider/cloudflare/cloudflare.go#L424-L427 are also redundant.